src/overlay.ts

```diff
@@ -0,0 +1,101 @@
+import { EOL } from 'os';
+
+// https://github.com/rescript-lang/rescript-vscode/blob/7ab2d231f91fee2f93cbf6cae1b38f94c06a58c1/server/src/utils.ts#L288
+const fileAndRangeRegex = /(.+):(\d+):(\d+)(-(\d+)(:(\d+))?)?$/;
+
+// https://github.com/rescript-lang/rescript-vscode/blob/7ab2d231f91fee2f93cbf6cae1b38f94c06a58c1/server/src/utils.ts#L433
+const codeRegex = /^  +([0-9]+| +|\.) (│|┆)/;
+
+// Compiler can treat warnings as hard errors based on `warnings` object in bsconfig.json
+const warningErrorRegex = /Warning number \d+ \(configured as error\)/;
+
+// Returns true if the line indicates the start of an error block
+function isErrorLine(line: string) {
+  if (line.startsWith("  We've found a bug for you!")) return true;
+  if (line.startsWith('  Syntax error!')) return true;
+  if (warningErrorRegex.test(line)) return true;
+  return false;
+}
+
+export default function parseErrorLog(log: string) {
+  // Split by line endings and remove empty lines
+  const lines = log.split(EOL).filter(Boolean);
+
+  // Optimization; only parse log when compiler is done
+  if (lines[lines.length - 1]?.startsWith('#Done(')) {
+    let foundError = false;
+    let path = '';
+    let startLine = 0;
+
+    // There can be additional messages such as hints
+    const messages = [];
+
+    // Avoid trimming the indentation in vite overlay by adding a horizontal ruler
+    // https://github.com/vitejs/vite/blob/96591bf9989529de839ba89958755eafe4c445ae/packages/vite/src/client/overlay.ts#L144
+    const frame = ['—'.repeat(80)];
```

Comment:
Is is possible to lift the `'—'.repeat(80)` to the top-level and initialize it once?

Reply:
6c04d59